pkg/pipeline/controller.go

```diff
@@ -405,7 +407,12 @@ func (c *Controller) SendEOS(ctx context.Context) {
 
 		case livekit.EgressStatus_EGRESS_ENDING,
 			livekit.EgressStatus_EGRESS_LIMIT_REACHED:
-			go c.p.SendEOS()
+			go func() {
+				c.eosTimer = time.AfterFunc(time.Second*30, func() {
+					c.OnError(errors.ErrPipelineFrozen)
+				})
+				c.p.SendEOS()
```

Comment:
Will this kill in process uploads? We have many examples of cases where upload takes more than 30 seconds (a 3h file could be several GB, and even for shorter sessions, we have cases where S3 is just slow). We should not charge customers for that time, but we should keep the upload going.

Reply:
the media won't be available if the eos is never received

Reply:
(unless it's HLS, in which case the playlist will not be closed)

Reply:
My question is about calling `c.OnError(errors.ErrPipelineFrozen)` after 30s.